flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/FlatMapRunner.scala

```diff
@@ -43,7 +45,15 @@ class FlatMapRunner[IN, OUT](
     LOG.debug(s"Compiling FlatMapFunction: $name \n\n Code:\n$code")
     val clazz = compile(getRuntimeContext.getUserCodeClassLoader, name, code)
     LOG.debug("Instantiating FlatMapFunction.")
-    function = clazz.newInstance()
+    if (null == parameterValues || null == parameterTypes ||
+      parameterValues.length == 0 || parameterTypes.length == 0 ||
+      parameterTypes.length != parameterValues.length) {
```

Comment:
I think we do not need to check `parameterValues.length == 0`, if the parameter length is 0, it also can be handled correctly by the `else` brach. And also, I think we should not to check `parameterTypes.length != parameterValues.length`, in this case, a compile exception thrown would be better rather than ignoring the parameters and using the 0-parameter constructor.

So I think `null == parameterValues || null == parameterTypes` is enough and simpler.